operator/pkg/apis/core/customize/v1beta1/controllerresource_types.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	addonv1alpha1 "sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/apis/v1alpha1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=controllerresources,scope=Cluster
+
+// ControllerResource is the Schema for resource customization API for config connector controllers.
+type ControllerResource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ControllerResourceSpec   `json:"spec"`
+	Status ControllerResourceStatus `json:"status,omitempty"`
+}
+
+// ControllerResourceSpec is the specification of the resource customization for containers of
+// a config connector controller.
+type ControllerResourceSpec struct {
```

Comment:
Nit (more of a general thought than a blocker): We don't have any way of enforcing that v1alpha1 == v1beta1 or that it round-trips (e.g. when we add a field here, will we remember to add it to v1alpha1?).  Perhaps we don't care about adding fields to v1alpha1 if it is unserved.

But should we include a round-trip test or another trick?  cc @apelisse

Reply:
I think you're supposed to do a round-trip test, but I don't know how easy we make it for people to do that. I always felt like if that's a requirement of the api, controller-runtime/kubebuilder should generate a test for you and run it.

Reply:
Thank you @apelisse, is there an example round-trip test you can point me to for reference?